compio-native-tls/src/stream.rs

```diff
@@ -0,0 +1,78 @@
+use std::{
+    io::{self, Read, Write},
+    mem::MaybeUninit,
+};
+
+use compio_buf::{BufResult, IoBuf, IoBufMut};
+use compio_io::{AsyncRead, AsyncWrite};
+
+use crate::StreamWrapper;
+
+fn filled(slice: &mut [MaybeUninit<u8>]) -> &mut [u8] {
+    slice.fill(MaybeUninit::new(0));
+    unsafe { std::slice::from_raw_parts_mut(slice.as_mut_ptr().cast(), slice.len()) }
+}
+
+/// A wrapper around an underlying raw stream which implements the TLS or SSL
+/// protocol.
+///
+/// A `TlsStream<S>` represents a handshake that has been completed successfully
+/// and both the server and the client are ready for receiving and sending
+/// data. Bytes read from a `TlsStream` are decrypted from `S` and bytes written
+/// to a `TlsStream` are encrypted when passing through to `S`.
+#[derive(Debug)]
+pub struct TlsStream<S>(native_tls::TlsStream<StreamWrapper<S>>);
+
+impl<S> From<native_tls::TlsStream<StreamWrapper<S>>> for TlsStream<S> {
+    fn from(value: native_tls::TlsStream<StreamWrapper<S>>) -> Self {
+        Self(value)
+    }
+}
+
+impl<S: AsyncRead> AsyncRead for TlsStream<S> {
+    async fn read<B: IoBufMut>(&mut self, mut buf: B) -> BufResult<usize, B> {
+        let slice = filled(buf.as_mut_slice());
+        loop {
+            let res = Read::read(&mut self.0, slice);
```

Comment:
Use `io::Read::read` to avoid ambiguity. Just `Read::read` looks like an OpCode.